File: api/src/main/java/com/beco/api/model/dto/CustomerDto.java
package com.beco.api.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.Instant;
import java.util.Date;
import java.util.UUID;

@Data
public class CustomerDto {

    private UUID customerId;

    private CompanyDto company;

    private ContactDto contact;

    private SharedDetailsDto sharedDetails;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date dateStart;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date dateEnd;

    private Boolean isSolvent = true;

    private CustomerDto parent;

    private Instant createdDate;

    private Instant lastModifiedDate;
}

File: spa/src/app/core/model/dto/customer.dto.ts
import { CompanyDto } from './company.dto';
import { ContactDto } from './contact.dto';
import { SharedDetailsDto } from './shared-details.dto';

export interface CustomerDto {
  customerId: string;
  company: CompanyDto;
  contact: ContactDto;
  sharedDetails: SharedDetailsDto;
  dateStart: string; // yyyy-MM-dd
  dateEnd: string;
  isSolvent: boolean;
  parent?: CustomerDto; // récursif, facultatif
  createdDate: string;
  lastModifiedDate: string;
}
